lib/dao/payment_dao.dart

```diff
@@ -103,4 +105,194 @@ class PaymentDao {
+      whereArgs: [9], // Assuming you want to filter for true
+    );
+    if (maps.isNotEmpty) {
+      return maps;
+    }
+    return null;
+  }
+
+  Future<Map<String,dynamic>?> searchForSameTitle(String title,int id) async{
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'title = ? AND id != ?',
+      whereArgs: [title,id], // Assuming you want to filter for true
+    );
+
+    if(maps.isNotEmpty){
+      return maps.first;
+    }
+
+    return null;
+  }
+
+
+  Future<List<Map<String,dynamic>>?> getAllMiscellanous() async{
+
+    List<Category> allCategories = await CategoryDao().find();
+    final db = await getDBInstance();
+    final List<Map<String, dynamic>> maps = await db.query(
+      'payments',
+      where: 'category = ?',
+      whereArgs: [10], // Assuming you want to filter for true
+    );
+
+    if(maps.isNotEmpty){
+      return maps;
+    }
+
+    return null;
+  }
+
+
+  Future<int> updateMiscellanousCategories() async{
+
+    int count = 0;
+    CategoryDao categoryDao = CategoryDao();
+    AccountDao accountDao = AccountDao();
+    List<Map<String,dynamic>> allMisPayments = await getAllMiscellanous() ?? [];
+    if(allMisPayments.isNotEmpty){
+
+      for(Map<String,dynamic> misPayment in allMisPayments){
+        Map<String,dynamic> gotPayment = await searchForSameTitle(misPayment['title'], misPayment['id']) ?? {};
+        if(gotPayment.isNotEmpty){
+          Category? cat = await categoryDao.findCategoryById(gotPayment['category']);
+          Account? acc = await accountDao.findCategoryById(misPayment['account']);
+          final updatedMap = {"id": misPayment['id'], "title": misPayment['title'], "description": misPayment['description'], "account": acc!.toJson(), "category": cat!.toJson(), "amount": misPayment['account'].toDouble(), "type": misPayment['type'], "datetime": misPayment['datetime'], "autoCategorizationEnabled": misPayment['autoCategorizationEnabled'] == 1 ? true : false};
+          await upsert(Payment.fromJson(updatedMap),);
+          count++;
+        }
+      }
+
+    }
+    return count;
+  }
+
+  Future<int> categorizeUsingRules()async{
+
+    CategoryDao categoryDao = CategoryDao();
+    AccountDao accountDao = AccountDao();
+    final db = await getDBInstance();
+    List<String> titles = [
+      "Toll charges",
+      "Motors",
+      "Food",
+      "Swiggy",
+      "Zomato",
+      "Bistro",
+      "Restaurant",
+      "pharmacy",
+      "Diagnostics",
+      "BookMyShow"
+    ];
+
+    Map<String,dynamic> mapOfCategory = SharedPreferncesHelper.getListOfCategoryRules();
+    List<String> getUserRules = SharedPreferncesHelper.getListOfUserRules();
+    titles.addAll(getUserRules);
+
+// Get User Defined Rules using below line
+// List userRules = await getUserRules();
+// titles.addAll(userRules);
+
+    String placeholders = List.generate(titles.length, (index) => '?').join(',');
```

Comment:
I am unable to understand why we need to use dummy data.
Try to avoid these. You can take help of chatGpt/Claude.